external/kafka-0-10-sql/src/main/scala/org/apache/spark/sql/kafka010/package.scala

```diff
@@ -16,9 +16,36 @@
  */
 package org.apache.spark.sql
 
+import java.util.concurrent.TimeUnit
+
 import org.apache.kafka.common.TopicPartition
 
+import org.apache.spark.internal.config.ConfigBuilder
+
 package object kafka010 {   // scalastyle:ignore
   // ^^ scalastyle:ignore is for ignoring warnings about digits in package name
   type PartitionOffsetMap = Map[TopicPartition, Long]
-}
+
+  private[spark] val PRODUCER_CACHE_TIMEOUT =
+    ConfigBuilder("spark.kafka.producer.cache.timeout")
+      .doc("The time to remove the producer when the producer is not used.")
+      .timeConf(TimeUnit.MILLISECONDS)
+      .createWithDefaultString("10m")
```

Comment:
Nit: I prefer type safety so I would use `createWithDefault()`, like

```
  createWithDefault(TimeUnit.MINUTES.toMillis(10))
```

Reply:
Doesn't it still result in the same config type? I might have asked for this change the other way a while ago

Reply:
Even though it's not always the custom in the code +1 on type safety as long as the type of the config is the same (double checked and remained the same with @attilapiros suggestion)